project/SparkBuild.scala

```diff
@@ -531,8 +535,16 @@ object Unidoc {
       "-noqualifier", "java.lang"
     ),
 
-    // Group similar methods together based on the @group annotation.
-    scalacOptions in (ScalaUnidoc, unidoc) ++= Seq("-groups")
+    scalacOptions in (ScalaUnidoc, unidoc) ++= Seq(
+      "-groups" // Group similar methods together based on the @group annotation.
+    ) ++ (
+      // Add link to github sources when generating scaladoc for a non-snapshot release
+      if (!isSnapshot.value) {
+        Opts.doc.sourceUrl(s"https://github.com/apache/spark/tree/v${version.value}€{FILE_PATH}.scala")
```

Comment:
I suspect this is fine if, y'know, it builds correctly and outputs the links you expect. Have you tried it?

The € is a typo, right?